### PR TITLE
Feature/issue 3313 log redact db package

### PR DESCRIPTION
**Pull request by tleyden**
Log redaction for the db package

### CODE REVIEW

File: db/event_handler.go [hunk 1]
```diff
@@ -92,7 +92,7 @@ func (wh *Webhook) HandleEvent(event Event) {
 		// for DocumentChangeEvent, post document body
 		jsonOut, err := json.Marshal(event.Doc)
 		if err != nil {
-			base.Warn("Error marshalling doc for webhook post")
+			base.WarnR("Error marshalling doc for webhook post")
```
Comment (bbrks) [inline db/event_handler.go]: Not strictly related, so feel free to ignore. But perhaps we could include the error in this log message, and the few below while we're here?

File: db/kv_dense_block.go [hunk 2]
```diff
@@ -218,8 +218,8 @@ func (d *DenseBlock) addEntry(logEntry *LogEntry) (changed bool, removalRequired
 		// Ensure this entry hasn't already been written by another writer
 		clockSequence := d.getClock()[logEntry.VbNo]
 		if logEntry.Sequence <= clockSequence {
-			base.LogTo("ChannelStorage+", "Index already has entries later than or matching sequence - skipping.  key:[%s] seq:[%d] index_seq[%d] blockKey:[%s]",
-				logEntry.DocID, logEntry.Sequence, clockSequence, d.Key)
+			base.LogToR("ChannelStorage+", "Index already has entries later than or matching sequence - skipping.  key:[%s] seq:[%d] index_seq[%d] blockKey:[%s]",
+				base.UD(logEntry.DocID), logEntry.Sequence, clockSequence, base.UD(d.Key))
```
Comment (bbrks) [inline db/kv_dense_block.go]: `d.Key` sometimes not redacted (e.g. line 161, 571, 582), but it looks like it's a SGW-generated value. Maybe no need for `base.UD` here?

**Review by bbrks — CHANGES_REQUESTED** [review]
Just a few notes, but overall looks good!

File: db/change_cache.go [hunk 3]
```diff
@@ -240,9 +240,9 @@ func (c *changeCache) CleanSkippedSequenceQueue() bool {
 					found = append(found, entries[0])
 				} else {
 					if err != nil {
-						base.Warn("Error retrieving changes from view during skipped sequence check:", err)
+						base.WarnR("Error retrieving changes from view during skipped sequence check:", err)
```
[the remark this answers is not in the document]
Comment (tleyden) [reply db/change_cache.go]: Fixed

File: db/database.go [hunk 4]
```diff
@@ -246,13 +246,13 @@ func NewDatabaseContext(dbName string, bucket base.Bucket, autoImport bool, opti
...
-		base.LogTo("Feed", "Starting mutation feed on bucket %v due to either channel cache mode or doc tracking (auto-import/bucketshadow)", bucket.GetName())
+		base.LogToR("Feed", "Starting mutation feed on bucket %v due to either channel cache mode or doc tracking (auto-import/bucketshadow)", base.UD(bucket.GetName()))
 
 		if err = context.tapListener.Start(bucket, options.TrackDocs, feedMode, func(bucket string, err error) {
 
-			msg := fmt.Sprintf("%v dropped Mutation Feed (TAP/DCP) due to error: %v, taking offline", bucket, err)
-			base.Warn(msg)
-			errTakeDbOffline := context.TakeDbOffline(msg)
+			msgFormat := "%v dropped Mutation Feed (TAP/DCP) due to error: %v, taking offline"
+			base.WarnR(fmt.Sprintf(msgFormat, base.UD(bucket), err))
```
[the remark this answers is not in the document]
Comment (tleyden) [reply db/database.go]: Fixed

File: db/database.go [hunk 5]
```diff
@@ -246,13 +246,13 @@ func NewDatabaseContext(dbName string, bucket base.Bucket, autoImport bool, opti
...
+			errTakeDbOffline := context.TakeDbOffline(fmt.Sprintf(msgFormat, "Bucket", err))
```
[the remark this answers is not in the document]
Comment (tleyden) [reply db/database.go]: Fixed

File: db/event_handler.go (hunk 1 above)
Comment (tleyden) [reply db/event_handler.go]: Fixed

File: db/kv_dense_block.go (hunk 2 above)
Comment (tleyden) [reply db/kv_dense_block.go]: Yeah, from what I could tell, the key here is just a pointer to the block document, and the key itself won't have any possibility of user data.  Removed this `base.UD()` call.

**Comment by tleyden** [comment]
Looks like I'll need to rebase this off the latest dev branch

**Comment by tleyden** [comment]
Rebased and did a force push.  Build in progress

**Review by bbrks — APPROVED** [review]

**Pull request merged**